airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -199,6 +216,25 @@ def _normalize_manifest(self) -> None:
             normalizer = ManifestNormalizer(self._source_config, self._declarative_component_schema)
             self._source_config = normalizer.normalize()
 
+    def _migrate_config(
+        self,
+        config_path: Optional[str],
+        mutable_config: MutableMapping[str, Any],
```

Comment:
we talked about the directly, Instead of taking in both `mutable_config` and `config`, its more straightforward to take in just `config` and perform the duplication into a mutable mapping inside this methodso it can be used by the `spec_component`'s migration and transformation methods. Then return the final config.